datafusion/sqllogictest/bin/sqllogictests.rs

```diff
@@ -16,57 +16,129 @@
 // under the License.
 
 use clap::Parser;
+use datafusion_common::instant::Instant;
 use datafusion_common::utils::get_available_parallelism;
+use datafusion_common::{exec_datafusion_err, exec_err, DataFusionError, Result};
+use datafusion_common_runtime::SpawnedTask;
 use datafusion_sqllogictest::{DataFusion, TestContext};
 use futures::stream::StreamExt;
+use indicatif::{
+    HumanDuration, MultiProgress, ProgressBar, ProgressDrawTarget, ProgressStyle,
+};
 use itertools::Itertools;
-use log::info;
-use sqllogictest::{strict_column_validator, Normalizer};
+use log::Level::{Info, Warn};
+use log::{info, log_enabled, warn};
+#[cfg(feature = "postgres")]
+use once_cell::sync::Lazy;
+use sqllogictest::{
+    parse_file, strict_column_validator, AsyncDB, Condition, Normalizer, Record,
+    Validator,
+};
+#[cfg(feature = "postgres")]
+use std::env::set_var;
 use std::ffi::OsStr;
 use std::fs;
+#[cfg(feature = "postgres")]
```

Comment:
Perhaps as a follow on PR the code that manages the postgres container could be moved into its own module (like `postgres_container.rs` or something so we only needed one `#[cfg(feature = "postgres")]`

I suspect this would also make the code a bit easier to reason about

Reply:
It is a bit of a mess eh? Yeah, I think that is a good idea